src/main/java/me/champeau/gradle/japicmp/ignore/processor/ClassProcessor.java

```diff
@@ -0,0 +1,72 @@
+package me.champeau.gradle.japicmp.ignore.processor;
+
+import japicmp.model.JApiClass;
+import japicmp.model.JApiCompatibilityChange;
+import japicmp.model.JApiImplementedInterface;
+import japicmp.model.JApiSuperclass;
+import me.champeau.gradle.japicmp.archive.VersionsRange;
+import me.champeau.gradle.japicmp.ignore.entity.EntityManager;
+import me.champeau.gradle.japicmp.ignore.entity.Provider;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ClassProcessor {
+  private final ClassMutator classMutator;
+  private final MethodProcessor methodProcessor;
+  private final FieldProcessor fieldProcessor;
+  private final ConstructorProcessor constructorProcessor;
+  private final EntityManager manager;
+  private final Provider.MutableProvider<JApiClass> provider = ProviderHelper.createDefaultClassProvider();
+
+  public ClassProcessor(EntityManager manager) {
+    this.manager = manager;
+    classMutator = new ClassMutator();
+    methodProcessor = new MethodProcessor(classMutator, manager);
+    fieldProcessor = new FieldProcessor(classMutator, manager);
+    constructorProcessor = new ConstructorProcessor(classMutator, manager);
+  }
+
+  public void process(List<JApiClass> classes, VersionsRange versions) {
+    for (JApiClass clazz : classes) {
+      doProcessClass(clazz, versions);
+      methodProcessor.process(clazz.getMethods(), versions);
+      fieldProcessor.process(clazz.getFields(), versions);
+      constructorProcessor.process(clazz.getConstructors(), versions);
+      classMutator.tryClearClass(clazz);
+    }
+  }
```

Comment:
It looks like it's not this class responsibility to own this method. `ClassProcessor` should just operate with class modifications, pretty much the same way methodsProcessor or fieldProcessor do.

Reply:
Done